rerouted all dependencies on guava collections from MPS.IDEA to de.q60.mps.collections.libs

enikao opened the pull request:
rerouted all dependencies in platfrom on com.google.common.collect(MPS.IDEA/) to com.google.common.collect(de.q60.mps.collections.libs/)

enikao commented:
equivalent to https://github.com/JetBrains/MPS-extensions/pull/1000

enikao commented:
Can you give me a hint what's failing? I don't have access to TeamCity, and local build fails because of gcc compilation issues.

alexanderpann commented:
You should be able to log in as a guest on TC. This build configuration is public: https://mps.builds.itemis.cloud/buildConfiguration/MbeddrEcosystem_Mbeddr_PullRequests/369899?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandPull+Request+Details=true&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=369899_524_482.5496&logFilter=debug&logView=flowAware

alexanderpann commented:
To give a hint: Check the solution test.com.mbeddr.mpsutil.ecore.metaModelImport in com.mbeddr.platform.ts.tests.build. It seems like it has some dependencies that get automatically removed when you update the build script. The same thing happened to me.

enikao commented:
> To give a hint: Check the solution test.com.mbeddr.mpsutil.ecore.metaModelImport in com.mbeddr.platform.ts.tests.build. It seems like it has some dependencies that get automatically removed when you update the build script. The same thing happened to me.

Thanks for the hint! That's a nasty one

enikao commented:
I can't get the build to pass. It now complains about 

```
Step 1/1: Check for dirty files (Command Line)

Dirty files detected after build/migration, run './gradlew build migrate remigrate' and/or update .gitignore to fix:
On branch premerge/369995
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
  modified:   code/languages/com.mbeddr.mpsutil/languages/com.mbeddr.mpsutil.logicalChild/com.mbeddr.mpsutil.logicalChild.mpl
  modified:   code/languages/com.mbeddr.mpsutil/solutions/com.mbeddr.mpsutil.ecore.runtime/com.mbeddr.mpsutil.ecore.runtime.msd
  modified:   code/languages/com.mbeddr.mpsutil/solutions/com.mbeddr.mpsutil.ecore.testing.runtime/com.mbeddr.mpsutil.ecore.testing.runtime.msd
  modified:   code/languages/com.mbeddr.mpsutil/solutions/com.mbeddr.mpsutil.ecore.ui.modelexporter/com.mbeddr.mpsutil.ecore.ui.msd
  modified:   code/languages/com.mbeddr.mpsutil/solutions/com.mbeddr.mpsutil.ecoretransformationutils.stubs/com.mbeddr.mpsutil.ecore.stubs.msd
  modified:   code/languages/com.mbeddr.mpsutil/tests/test.com.mbeddr.mpsutil.common/test.com.mbeddr.mpsutil.common.msd
```

These files don't change if I run it locally in MPS, and I cannot run the proposed command (on Windows fails with no make available, on WSL fails with gcc compilation issues).

Any hints how to get there?

alexanderpann commented:
Did you try building those models, call force save all and check the exact MPS version? I know it can be tough to get it working.
There should be some output starting with >>>> and ending with <<<< that is the diff itself. I normally just copy it to a file MIGRATE, strip the log output and apply the diff:
`sed 's/^.\{13\}//' /Users/apann/Desktop/MIGRATE | git apply`
Alternatively, you could modify the migrate and remigrate task to not build the project before.

enikao commented:
Yay, success! Checking out _again_ in WSL (not reusing the Windows directory) worked.

alexanderpann reviewed (approved):
LGTM. Thanks for making the change!
